### ISSUE TITLE
phased array radar model

### ISSUE THREAD

**Issue by zrisher** [Rynchodon/Autopilot#16]
I've been looking through various radar designs and it seems like these are most commonly used in space avionics and modern military roles due to their ability to track multiple targets and redundancy of detective surfaces:

http://intercepts.defensenews.com/wp-content/uploads/2014/11/0012-DSC_4632b.jpg
http://media.defenceindustrydaily.com/images/ELEC_CEA_Radars_Active_on_ANZAC_Concept_lg.jpg
http://www.nato.int/nato_static/assets/pictures/2012_12_121203b-patriot/20121204_121121a-014.jpg
https://www.ll.mit.edu/news/images/MPARillustration.jpg
http://www.latinaero.com/news/defence/2012_10/images/nws_dfc_121002_01-003.jpg
http://www.airforce-technology.com/projects/mig-31/images/2-mig-31-foxhound.jpg

Plus if you use one one the all-sided ones it never has to rotate.

I can't help loving the current radar model because of what it allows me to do, but it does look and act a bit like a placeholder. If we ever revisited the model, we should consider using a phased array design.

**Comment by zrisher** [comment]
This: http://en.wikipedia.org/wiki/Active_electronically_scanned_array

**Comment by Rynchodon** [comment]
I did do research before building the model. I went with a design the people would recognize as "radar" rather than a realistic-looking one. 
I figure most people would not be able to identify these:
* [TPS-77] (http://www.google.com/imgres?imgurl=http://www.lockheedmartin.com/content/dam/lockheed/data/ms2/photo/tps-77/TPS-77-radar-05.jpg&imgrefurl=http://www.lockheedmartin.com/us/news/press-releases/2012/november/lockheed-martin-and-pt-cmi-teknologi-complete-radar-production-r.html&h=980&w=1200&tbnid=A9SG8C7v2JByWM:&zoom=1&docid=RbGGTKRoHgh4nM&ei=h4QNVc-_EOTLsATs14C4Bw&tbm=isch&ved=0CDcQMygvMC84rAI)
* [Radar Dome] (http://www.google.com/imgres?imgurl=http://pixabay.com/static/uploads/photo/2012/11/28/09/17/radar-dome-67536_640.jpg&imgrefurl=http://pixabay.com/en/radar-dome-antennas-measurement-67536/&h=414&w=640&tbnid=vZlptQBfgMdVtM:&zoom=1&docid=2tIZvIVECvJN2M&itg=1&ei=RYMNVZqLNYO1sQTg9YGoBg&tbm=isch&ved=0CF8QMyg8MDw)
* [AESA] (http://www.google.com/imgres?imgurl=http://www.defense.gouv.fr/var/dicod/storage/images/base-de-medias/images/dga/1_programmes/le-radar-rbe2-a-antenne-active-qui-equipe-le-rafale/1994801-1-fre-FR/le-radar-rbe2-a-antenne-active-qui-equipe-le-rafale.jpg&imgrefurl=http://defensetech.org/2012/10/04/dassault-delivers-aesa-radar-equipped-euro-fighter/&h=526&w=800&tbnid=dIkpjTOY2Fa3yM:&zoom=1&docid=Y1jRu0_7NU7JVM&ei=K4QNVdayK-rlsAS5woCYBQ&tbm=isch&ved=0CEEQMyg5MDk4yAE)
* [Recon Aircraft] (http://www.google.com/imgres?imgurl=http://pixabay.com/static/uploads/photo/2012/10/26/01/11/radar-62991_640.jpg&imgrefurl=http://pixabay.com/en/radar-aircraft-squadron-62991/&h=425&w=640&tbnid=jToZEv7eo83IOM:&zoom=1&docid=wPzeLZc2drf-UM&ei=RYMNVZqLNYO1sQTg9YGoBg&tbm=isch&ved=0CEoQMygnMCc)  

My plan was to make a model, have someone say "I can do better than that.", and then I get a better one. 
The script also allows for multiple designs to be used, or blocks from other mods to work.

**Comment by zrisher** [comment]
Totally makes sense.  "I would like to do better than that, but it'll take me a while since I have no experience with modeling" is where I'm at. But eventually I will definitely jump on this.

Thinking about this some more, you could add some nice strategic gameplay decisions by providing different types of radar blocks:
* Rotating simple parabolic dish arrays are cheap and have big range, but are easy to disable and/or jam, track slowly, can only track a single target at once, and provide a big radar signature
* Phased arrays are expensive (have to put one on each side of your ship too), have less range, but provide multiple target tracking, much faster tracking, a smaller radar signature, and a surface that can take a beating and still work
* Active phased arrays are just like phased arrays but more expensive, can track more targets, even harder to jam, and a small radar signature

There's some extra complexity there, but nothing Autopilot can't handle with the right systems in place I think. Happy to take a stab at this stuff after harvesting.

**Comment by Souper07** [comment]
I'm  starting to feel like radar should be a seperate mod from autopilot. Autopilot works just fine for engaging enemy ships just by using an antenna. You would only need a radar dish if the enemy isn't broadcasting  or you are out hunting for unpowered exploration ships.  Sure they could go hand-in-hand when both installed but I feel like radar should get its own mod, where it can be made more complex and functional on its own level with multiple types and blocks. It isn't necessarily a part of an autopilot system ,unless you want to rename the mod to Additional Cockpit Enhancements or Autopilot + Radar or something like that.  Thoughts?

**Comment by Rynchodon** [comment]
I do not really feel like Radar is a part of Autopilot, so much as Autopilot requires Radar. I cannot see `E` or `M` being useful at all, in multiplayer, without radar. 
Space Engineers does not offer a simple way for mods to interact, they can only send messages back and forth using chat, so separating Radar would have a performance hit.  
I am open to changing the name of Autopilot, but it is not something that should be done lightly.

**Comment by zrisher** [comment]
Was just going to say the same thing - you can't really `E` or `M` without radar. And radar implemented with that level of complexity would rely a lot on autopilot in its implementation.

Plus, you can always just edit the settings file to disable radar if you don't want it.

Now changing the name of Autopilot, maybe eventually that makes sense as this mod now encompasses three moderately separate functional sets. But it would be much easier to wait until functionality matures and make that decision then.

**Comment by zrisher** [comment]
I feel like an appropriate name would convey that this mod provides "Electronic Navigation, Communication, and Targeting Systems", but I can't think of a simple way to say that.

**Comment by Rynchodon** [comment]
Super cool awesome sauce?

**Comment by Souper07** [comment]
+1 make it so

**Comment by Souper07** [comment]
Is this still on the table, having different types of radar?

**Comment by zrisher** [comment]
Eventually I plan on working on it when I clear a few other things off my plate, if Rynch hasn't already accomplished before then.

**Comment by Souper07** [comment]
Ok, I'm currently working updating Rynchodon's radar dish with some newer textures/construction stages, and a better collision box. After that, I wouldn't mind making some more radar models, if someone will do something with them.

**Comment by zrisher** [comment]
Yes, I would definitely make use of them whenever I get around to working on this, which is "definitely eventually" : ). I'd like to have 3 tiers, as mentioned above - parabolic, passive phased array, active phased array. Parabolic would look generally as is now but with the visual updates you mentioned. Passive and active phased array would be large flat circles you put on a surface that detect in an 180* arc, basically just like turrets. I was thinking a small (1x1 or 2x2) and large (maybe 8x8 or bigger?) version with different ranges, ala the images above.

**Comment by Rynchodon** [comment]
>After that, I wouldn't mind making some more radar models, if someone will do something with them.

Other than simple changes to the formula, implementing different radar would take some work and I still haven't done the jump drive yet (or the ore detector, or the...).

**Comment by zrisher** [comment]
@Rynchodon Yeah, I'm happy to jump on this when I can, expect to in the next couple months.

**Comment by Rynchodon** [comment]
I have started on a better Radar script, so far the plan includes:
* Radar can be any type of fixed block
* Angle restrictions
* Obstruction tests
* Decoys
* Radar jamming (radar can be resistant)
* Passive detection of other radars and/or radar jammers

I considered a maximum number of targets. It would an enormous pain to code cooperation between radars. Between angle restrictions and obstruction testing, it may be necessary to build multiple radars anyway.

Is there anything that I have missed?

**Comment by zrisher** [comment]
Only other things I can think of:
* Maximum number of targets a block can track - would definitely be good, no need for managed cooperation between radars as long as the block tracks the closest targets. You don't need `Block A can detect 3 and Block B can detect 5 so we can detect 8` so much as `Block A can only detect a few but Block B can detect a bunch so I want that eventually`.
* Radar tech levels - Affects ability to accurately judge objects at a a given range, susceptibility to jamming from a given level, and radar signature
* Jamming tech levels - Affects ability to jam radar of a given level and radar signature

For the tech levels, radar signature and accuracy could be hardcoded into the block easily enough, but you still want some sort of `Radar level X vs Jamming level Y` dynamic that I think would be easiest to do with a levels system.

**Comment by zrisher** [comment]
Oh and two more things, but I don't really expect this to ever get these. It would be awesome but sounds like a ton of work:
* Radar signatures based on total surface exposure to the source of the beam. So you can build a ship with less big flat surfaces and it would have a lower radar signature.
* Material-based signatures - you can build out [Radar Absorbent Materials](https://en.wikipedia.org/wiki/Radar-absorbent_material) and get a smaller signature

**Comment by Rynchodon** [comment]
I don't care for maximum number of targets but I'll add the basic support you requested.

Radar / Jamming tech levels: I don't really know what you are asking for here, the jammers can have different strengths and the radars can have different resistances. I have not decided exactly how these values would interact yet.

>Radar signatures based on total surface exposure to the source of the beam. So you can build a ship with less big flat surfaces and it would have a lower radar signature.

Making the signature based on the ship's profile is doable, but I doubt many players would care when it would be so much less work to build a jammer.

**Comment by Rynchodon** [comment]
All the properties of a radar can be specified through the .sbc files; the blocks will not need to be included in Autopilot to work.

**Comment by Rynchodon** [comment]
I have not tested everything but it is all together: https://github.com/Rynchodon/Autopilot/blob/Radar/AntennaRelay/Scripts/Radar.cs

I decided to restrict radars and jammers to be beacons or antenna, I can always switch it back.

@Souper07 Are you still interested in creating (a) radar block(s)?

**Comment by zrisher** [comment]
First of all, looks cool! All of the below are suggestions that could be implemented down the line and I don't expect any of them immediately. : )

* `Obstructed()` gets all the entities from API Gateway, but it's called right after doing the same thing in `ActiveDetection` (though `PassiveDetection` is able to use the established `AllRadarAndJam` list). Perhaps we could optionally pass an existing entity list in to save it from having to do that?
* PassiveDetection - would this also work on other antennae or beacons that aren't radar? Just based on some static signal strength for those blocks * their range?
* Radar/Jamming tech levels - Looking at what you've done and thinking about it, this seems unnecessary. Can probably model it into this simpler system. Only change that might be nice is for the jamming & detection to not be a static definite yes/no but rather have a particular % chance of occurring depending on the strengths involved.This would give you the cool radar "blipping" characteristic of an object moving between the range at which it's perfectly jammed  / not detected to the range where it's always detected as it approaches its target. Maybe damaged radar blocks would occasionally see ghost signals in a similar manner as well.
* HUD markers would still be cool
* Materials-based signatures still would be cool, but the info would need to be aggregated at the grid level already for these update calls to not be massively time consuming right? Unless we had a entity logic component on each grid that read the properties of added blocks for a "radar absorption" attribute.
* Cross-section instead of volume detection - 
  * "Making the signature based on the ship's profile is doable, but I doubt many players would care when it would be so much less work to build a jammer."
  * True, but what if all the jammers put off enough of a passive signal to be useless at hiding themselves, mostly useful for hiding fleet size? Or if the jammers are large and energy intensive, not appropriate to small strike fighters that might benefit from radar sneakiness?

**Comment by Rynchodon** [comment]
>* Obstructed() gets all the entities from API Gateway, but it's called right after doing the same thing in ActiveDetection (though PassiveDetection is able to use the established AllRadarAndJam list). Perhaps we could optionally pass an existing entity list in to save it from having to do that?

Perhaps I should just keep a collection of all the entites, so I never actually have to use this.

>* PassiveDetection - would this also work on other antennae or beacons that aren't radar? Just based on some static signal strength for those blocks * their range?

I supposed it could be expanded to include anything, I have no idea where you are going with this.

>* Radar/Jamming tech levels - Looking at what you've done and thinking about it, this seems unnecessary. Can probably model it into this simpler system. Only change that might be nice is for the jamming & detection to not be a static definite yes/no but rather have a particular % chance of occurring depending on the strengths involved.This would give you the cool radar "blipping" characteristic of an object moving between the range at which it's perfectly jammed / not detected to the range where it's always detected as it approaches its target. Maybe damaged radar blocks would occasionally see ghost signals in a similar manner as well.

The position prediction would kinda defeat this, especially for objects in uniform linear motion.

>* HUD markers would still be cool

I'm working on it, it's not part of radar though. I have no quite worked out the details for how a player would configure it, I guess slash commands are in order.

>* Materials-based signatures still would be cool, but the info would need to be aggregated at the grid level already for these update calls to not be massively time consuming right? Unless we had a entity logic component on each grid that read the properties of added blocks for a "radar absorption" attribute.

It should be doable as long as there is a default that most blocks fall under. If only block counts are needed, CubeGridCache could suffice. 

>* Cross-section instead of volume detection -
  * "Making the signature based on the ship's profile is doable, but I doubt many players would care when it would be so much less work to build a jammer."
  * True, but what if all the jammers put off enough of a passive signal to be useless at hiding themselves, mostly useful for hiding fleet size? Or if the jammers are large and energy intensive, not appropriate to small strike fighters that might benefit from radar sneakiness?

I thought you wanted the game to run faster?

**Comment by zrisher** [comment]
> Perhaps I should just keep a collection of all the entites, so I never actually have to use this.

Yeah I think that would make sense for this purpose. You could even maintain your own Trees for doing bounding checks and raycasts. That's what we do in GP now: 

https://github.com/zrisher/SEGarden/blob/master/Math/AABBTree.cs
https://github.com/zrisher/SEGarden/blob/master/Math/AABBEntity.cs

> I supposed it could be expanded to include anything, I have no idea where you are going with this.
Just the fact that radios put off electromagnetic signals that can be picked up by passive radar. Obviously not a big priority. Would just be kind of cool.

> I'm working on it, it's not part of radar though. I have no quite worked out the details for how a player would configure it, I guess slash commands are in order.

I'm happy to incorporate SEGarden if you like in a PR - it's a fair amount of code, and it does run its own update manager (adapted from this one), but obviously it doesn't do much if not in use. That has a built in command model that makes these pretty easy, see:

https://github.com/zrisher/GardenPerformance/blob/master/Data/Scripts/GardenPerformance/Concealment/Commands.cs
https://github.com/zrisher/SEGarden/tree/master/Chat/Commands

> I thought you wanted the game to run faster?

Yesssss you're right, it would be pretty intensive to check. I just like being the voice that mentions every possible use case. : )

**Comment by Rynchodon** [comment]
>Yeah I think that would make sense for this purpose. You could even maintain your own Trees for doing bounding checks and raycasts. That's what we do in GP now:

I'll leave that level of optimization to people who know what they are doing.

>Just the fact that radios put off electromagnetic signals that can be picked up by passive radar. Obviously not a big priority. Would just be kind of cool.

Radio antennas? You want to detect them beyond their broadcasting range or something?

>I'm happy to incorporate SEGarden if you like in a PR - it's a fair amount of code, and it does run its own update manager (adapted from this one), but obviously it doesn't do much if not in use. That has a built in command model that makes these pretty easy, see:...

Easy like rocket science, maybe it will make more sense in the morning.

>Yesssss you're right, it would be pretty intensive to check. I just like being the voice that mentions every possible use case. : )

It's probably not that bad, PathChecker does this a lot, I just like to complain. I'm definitely not happy  with using volume. No idea how I could combine this with R.A.M.

**Comment by zrisher** [comment]
> Radio antennas? You want to detect them beyond their broadcasting range or something?

Yes the idea is the passive radar would pick up the position of the signal origins beyond their normal broadcast range as because it's specifically tuned to listen across the em spectrum without being specialized for communication. But like I said not a big deal.

> Easy like rocket science, maybe it will make more sense in the morning.

Lmk if you want to discuss.

**Comment by Souper07** [comment]
I've started working on some new models. When I first started this one I figured it could be some sort of radar jammer, but it could also be used as some sort of radar. You guys can do whatever you want with it. 
![capture](https://cloud.githubusercontent.com/assets/11542034/9780422/5a49ee24-573d-11e5-91b8-1177553ce78b.PNG)

**Comment by Souper07** [comment]
![dish2](https://cloud.githubusercontent.com/assets/11542034/9782088/284c242a-574f-11e5-8eac-9dc6f6e45ada.PNG)
![dish](https://cloud.githubusercontent.com/assets/11542034/9782090/2982c4a2-574f-11e5-9761-81ce732149a2.PNG)

**Comment by Souper07** [comment]
Some screens of the first finished model

![2015-09-14_00013](https://cloud.githubusercontent.com/assets/11542034/9866364/61ad965c-5b16-11e5-9c46-fe1c46df9dfc.jpg)
![2015-09-14_00012](https://cloud.githubusercontent.com/assets/11542034/9866365/629d6894-5b16-11e5-8722-2d050fb0345c.jpg)
![2015-09-14_00011](https://cloud.githubusercontent.com/assets/11542034/9866366/6394e880-5b16-11e5-8690-dfe20270c41a.jpg)
![2015-09-14_00010](https://cloud.githubusercontent.com/assets/11542034/9866367/64bec55a-5b16-11e5-922f-b8b62d3b86a2.jpg)
![2015-09-14_00009](https://cloud.githubusercontent.com/assets/11542034/9866371/669f6f78-5b16-11e5-921b-f0ba5d3b41a2.jpg)

**Comment by zrisher** [comment]
For me personally, the big white ball looks out of place. It seems like it should be armored instead of built out of a translucent material. But that's just me, I always think near-future military-ready, but all people have different views of this game.

**Comment by Souper07** [comment]
The white ball was inspired by the "Radar Dome" picture above. It could easily be changed to be a grey color, like the  ship in the first picture.

**Comment by Rynchodon** [comment]
I have to ask, can it tell the future?


>It seems like it should be armored instead of built out of a translucent material.

Wouldn't an armoured radar be counter-productive?

**Comment by Souper07** [comment]
Ha! No but it does grant wishes. How many radars/jammers are we talking here?

**Comment by zrisher** [comment]
> Wouldn't an armoured radar be counter-productive?

This stuff seems pretty strong and has very low EM interference:
http://www.dsm.com/content/dam/dsm/dyneema/en_GB/Downloads/Brochures/Brochure_Radomes%20with%20Dyneema.pdf

> How many radars/jammers are we talking here?

I was hoping to eventually get one low tier (single rotating dish, looks like the current radar, maybe smaller), one mid tier (looks like your new model), one standard high tier omni-directional (looks like one of the three panels visible at the top of [this ship](http://intercepts.defensenews.com/wp-content/uploads/2014/11/0012-DSC_4632b.jpg)) and one gigantic high tier omni-directional (e.g. [this](https://www.google.com/imgres?imgurl=http://mostlymissiledefense.files.wordpress.com/2012/04/fps-85image.jpg&imgrefurl=http://mostlymissiledefense.com/2012/04/12/the-fps-85-radar-april-12-2012/&h=1125&w=1500&tbnid=hjh9cUM2MqqL1M:&docid=mJzyTcWTH92xnM&ei=UbH4VemxJY2iyASt04_QBg&tbm=isch&ved=0CDUQMygMMAxqFQoTCKmerv2f-scCFQ0RkgodrekDag) or [this](https://www.google.com/imgres?imgurl=https://upload.wikimedia.org/wikipedia/commons/2/21/SSPARS_radar,_Clear_AFB.JPG&imgrefurl=https://en.wikipedia.org/wiki/Clear_Air_Force_Station&h=1360&w=2048&tbnid=qt85UOnUVIYv3M:&docid=GI7g92asXRLCxM&ei=UbH4VemxJY2iyASt04_QBg&tbm=isch&ved=0CCoQMygBMAFqFQoTCKmerv2f-scCFQ0RkgodrekDag)).

But the way Rynch is developing this allows radar blocks to be defined in other mods. So if this list is too exhaustive for Autopilot, I could fill in the gaps in another mod.

**Comment by Rynchodon** [comment]
>Ha! No but it does grant wishes.

I wish, I wish, for multiplayer fixes.

>How many radars/jammers are we talking here?

As many as you want to make.
I don't know how many radars or jammers I want to include in Autopilot itself. Maybe the best option is to put all the ones you make into one or more separate mods.

**Comment by Souper07** [comment]
Ok, so should I still PR to autopilot or are we going to setup a seperate project for this? The wish granting orb is just needing construction stages (had to start over :frowning: ) and then it is done. I've set up just a basic sbc file for it with the construction components all set to 1.

**Comment by Souper07** [comment]
>I wish, I wish, for multiplayer fixes.
How about some giant asteroids that will take months and months to develop.

**Comment by Rynchodon** [comment]
>Ok, so should I still PR to autopilot or are we going to setup a seperate project for this?

We can do a github project if you like or I can just tell you what to add to the .sbc files.
It's just a little section like:
<Description>
Radar
MaxTargets_Tracking=100
PassiveDetect_Radar=20000
SignalEnhance=1.1
</Description>

**Comment by zrisher** [comment]
I definitely would not mind these going straight into [GardenWarfare](https://github.com/zrisher/gardenwarfare). : ) I can commit to keeping that release up to date so people can use this feature of Autopilot. But also completely understand if it should be in an autopilot-support mod of its own.

**Comment by Rynchodon** [comment]
>I definitely would not mind these going straight into GardenWarfare. 

What, exactly, is your thinking behind this?

**Comment by zrisher** [comment]
> I don't know how many radars or jammers I want to include in Autopilot itself. Maybe the best option is to put all the ones you make into one or more separate mods.

and

> Ok, so should I still PR to autopilot or are we going to setup a seperate project for this?

 and

> We can do a github project if you like or I can just tell you what to add to the .sbc files.

suggest you were considering putting the additional implementing models into a separate mod to keep the options in autopilot simple. 

My thinking is I'm planning to release that mod anyway and it will be heavily dependent on autopilot, so it's an option for above considerations.

**Comment by Rynchodon** [comment]
>>I definitely would not mind these going straight into GardenWarfare.
>
What, exactly, is your thinking behind this?

Sorry, I had GardenWarfare confused with GardenConquest, you need to stop naming things garden.

**Comment by zrisher** [comment]
I have the shared namespace for brand awareness. : )

**Comment by Souper07** [comment]
"Phased Array" Radar
Still needs some detail work.
![2015-09-19_00001](https://cloud.githubusercontent.com/assets/11542034/9978892/97cc3aea-5f03-11e5-9d58-9b7388280336.jpg)
![2015-09-19_00002](https://cloud.githubusercontent.com/assets/11542034/9978893/993fcac2-5f03-11e5-9a1f-2b75a2d2f7de.jpg)

**Comment by Souper07** [comment]
For size comparison
![Uploading 2015-09-19_00003.jpg…]()

**Comment by Souper07** [comment]
For size comparison

![2015-09-19_00003](https://cloud.githubusercontent.com/assets/11542034/9978986/c15e716c-5f07-11e5-86eb-de5959c8ae5b.jpg)

**Comment by zrisher** [comment]
Looks awesome. Only thing I would change is maybe add some sort of protective cover over the honeycomb. And possibly make it a circular shape on top of its parallelogram mounting point. But I would be happy as-is.

Did you start with the construction stages this time? : )

What's the size of these in blocks?

**Comment by Rynchodon** [comment]
Any thoughts on the specs for the radars?

**Comment by zrisher** [comment]
My suggested versions would be:

Dish radar:
* Detects 1 object (no difference between large/small)
* Large has about 3x the power of the small
* Largest range per build cost of all the radars - primary reason to build these even late tech
* Very easy to jam
* Creates a large passive EM signature

Dome radar:
* 180 degree viewing (same as a regular turret. This shouldn't be able to detect objects below it, it has a mounting point and I'm not sure how you'd reliably accomplish 360 degree rotation on a parabolic radar dish)
* Detects 1 object
* Fairly easy to jam
* Basically we treat this as the same thing as the exposed dish, but it's able to rotate on its own inside the dome, and the dome provides some protection from electromagnetic interference so it's more sensitive (more targets, slightly harder to jam). It's the parabolic dish with all the bells and whistles.
* Slightly smaller total power output than the naked dishes

Passive Phased:
* Detects up to 5 objects
* Less overall radar power than the parabolic arrays, 
* Significantly smaller passive EM signature
* Moderately difficult to jam
* Can detect via EM device signatures
* Can take a beating and still work
* Second most expensive to build
* Fairly low power consumption

Active Phased (new category, can have the same model as passive phased, though we could add some cool lights or something to differentiate it)
* Detects up to 10 objects
* Most difficult to jam
* Smallest power consumption per radar power
* Very small EM signature

So basically there's two types - parabolic and phased, with two levels each. Parabolic are cheaper and have greater range. Phased have more sophisticated detection, are harder to detect themselves, and can track more targets.

**Comment by Rynchodon** [comment]
I don't know about you but I always have more than one ship. If the maximum objects is one, I'm just going to be detecting one of my ships most of the time.

>Dish radar:
> * Large has about 3x the power of the small

You want to boost the power of the small one? As previously discussed it is currently less than 1/5 the power.

>Dome radar:
>* 180 degree viewing (same as a regular turret.

??? Turrets have 360° of azimuth and 124° of elevation (despite what it says in data files). I have no idea where you got 180° from or what you mean by "viewing".

> This shouldn't be able to detect objects below it,

I thought about this, I will probably add a restriction but whatever you mount the block on is going to block the radar anyway.

> * Slightly smaller total power output than the naked dishes

Sorter range?

>Passive Phased:
> * Less overall radar power than the parabolic arrays,

Even shorter range, I don't like where this is going the next one is going to be like three feet.

> * Can detect via EM device signatures

That's what I said

> * Fairly low power consumption

I can't change power consumption

**Comment by zrisher** [comment]
> I don't know about you but I always have more than one ship. If the maximum objects is one, I'm just going to be detecting one of my ships most of the time.

Shouldn't it try to show enemy ships in descending order of size before showing your own?

> You want to boost the power of the small one? As previously discussed it is currently less than 1/5 the power.

Ok, that works for me!

> ??? Turrets have 360° of azimuth and 124° of elevation (despite what it says in data files). I have no idea where you got 180° from or what you mean by "viewing".

Was referring to elevation.

> Sorter range?

Yes, by lower output I meant shorter range / detection power.

> Even shorter range, I don't like where this is going the next one is going to be like three feet.

If we discuss the range as percentages of the largest one, the scale I was referring to wasn't 100-50-2-1, more like 100 - 95 - 70 - 70

> That's what I said

Yeah, wasn't writing these as diffs from your suggestions, I repeated attributes to make it complete on its own without reading the original, just for the sake of future readers.

> I can't change power consumption

Ahhhhh that sucks! I see `<RequiredPowerInput>0.0003</RequiredPowerInput>` on the autopilot block sbc, is it just this particular type of block we can't adjust power input for?

**Comment by Souper07** [comment]
>
>>
>>>
>>>>

**Comment by Souper07** [comment]
Radar Jamming Device
![2015-11-06_00006](https://cloud.githubusercontent.com/assets/11542034/11013910/aef9b4e8-84d5-11e5-8e94-3f3faefe944f.jpg)

**Comment by Rynchodon** [comment]
You're really on fire

**Comment by Rynchodon** [comment]
Jump'n Shootman created an AWACS radar for ARMS, any thoughts on balance? The block is 5x5x1.

![2016-01-14-20-06-13-350](https://cloud.githubusercontent.com/assets/11451518/12342667/8d73af58-bafa-11e5-983f-1c7599d78eef.png)

**Comment by Souper07** [comment]
I skimmed through this page https://en.wikipedia.org/wiki/Airborne_early_warning_and_control to get ideas. Maybe make it only able to detect ships but not stations. Maybe moving objects only?

**Comment by zrisher** [comment]
Main point of airborne radar is better detection range (vs ground-based) and mobility. Since we don't know if radar are mounted in space, in air, or on land, all the existing blocks can be mobile or static,  and we don't do any sort of power adjustments based on those things, the only differentiation between this block and the others is its size and shape. Give what I see there I'd make it long range and unidirectional.

**Comment by Rynchodon** [comment]
OK, so I think I'll make it a 30 km / 12 km range directional radar. I'm also going to rebalance and increase the costs of all radar equipment.
